conda_forge_tick/auto_tick.py

```diff
@@ -272,15 +242,11 @@ def add_replacement_migrator(
     total_graph = copy.deepcopy(gx)
 
     for node, node_attrs in gx.nodes.items():
-        meta_yaml = node_attrs["payload"].get("meta_yaml", {}) or {}
+        requirements = node_attrs["payload"]["requirements"]
+        rq = set().union([requirements[k] for k in ["build", "host", "run", "test"]])
         pkgs = {old_pkg}
-        old_pkg_c = len(pkgs.intersection(get_requirements(meta_yaml))) > 0
+        old_pkg_c = pkgs.intersection(rq)
 
-        rq = _host_run_test_dependencies(meta_yaml)
-
-        for e in list(total_graph.in_edges(node)):
-            if e[0] not in rq:
-                total_graph.remove_edge(*e)
```

Comment:
I see this block is removed here but not in the rebuild migrator below. Why is this?

Reply:
This motif is removing edges that are part of the wrong requirements section. In the rebuild we don't care about build, only host, run and test (assuming all feedstocks have the build/host split). So we don't want any edges from build, since that would cause weird rebuilds of build tooling that we don't need. This isn't needed in the replace package chunk because nominally all parts of the requirements are fair game, so no edges need removal. This is also why the arm rebuild doesn't have that section, it doesn't care if you are a build tool or not.

Reply:
Nominally we should replace these by making a multi graph with each edge labeled by the section it is in, but I didn't want to do that in this PR